torch/csrc/autograd/variable.h

```diff
@@ -238,12 +240,17 @@ struct TORCH_API AutogradMeta : public c10::AutogradMetaInterface {
     return grad_;
   }
 
+  const Variable& fw_grad(uint64_t level, const Variable& self) const override;
```

Comment:
Why does this take a self argument?

Reply:
This is needed as an argument for both this function and the set function because in the case where we need to replay a view, we need the metadata associated with self.